### PR TITLE
CLI: introduce `bb add` command for adding dependencies

### PR DESCRIPTION
**Pull request by siggisim**
**Related issues**: N/A

### CODE REVIEW

**Review by bduffany — APPROVED** [review]
nice :)

File: cli/add/add.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	// TODO(siggisim): Support MODULE.bazel
+	f, err := openOrCreateWorkspaceFile()
+	defer f.Close()
+
+	contents, err := io.ReadAll(f)
+	if err != nil {
+		return 1, err
+	}
+
+	versionKey := fmt.Sprintf("[https://registry.build/%s#%s]", module, resp.LatestReleaseWithWorkspaceSnippet)
```
[the remark this answers is not in the document]
Comment (siggisim) [reply cli/add/add.go:76]: I like it, but I'll need to update the registry.build urls to handle that (right now it's a hash param). Updated here.

File: cli/add/add.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+	flags = flag.NewFlagSet("add", flag.ContinueOnError)
+	usage = `
+usage: bb ` + flags.Name() + ` rules_go
+
+Adds the given dependency to your WORKSPACE file.
+`
+	headerTemplate = "###### Begin auto-generated section for %s ######"
+	footerTemplate = "###### End auto-generated section for %s ######"
+
+	headerRegex = regexp.MustCompile("##### Begin auto-generated section for \\[https:\\/\\/registry\\.build\\/([^#]+)#([^\\]]+)")
```
[the remark this answers is not in the document]
Comment (siggisim) [reply cli/add/add.go]: Done

File: cli/add/add.go [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+	if err != nil {
+		f, err := os.OpenFile(backupFileName, os.O_CREATE|os.O_APPEND|os.O_RDWR, 0644)
+		if err != nil {
+			return nil, err
+		}
+		workspacePath, err = os.Getwd()
+		if err != nil {
+			return nil, err
+		}
+		f.WriteString(`workspace(name = "` + path.Base(workspacePath) + `")` + "\n")
```
[the remark this answers is not in the document]
Comment (siggisim) [reply cli/add/add.go]: Done

File: cli/add/add.go [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+		f.Close()
```
[the remark this answers is not in the document]
Comment (siggisim) [reply cli/add/add.go]: Done (pulled it out into its own function and did the defer)

File: cli/add/add.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (siggisim) [reply cli/add/add.go]: Updated the registry to support these @ delimited paths

**Pull request merged**